src/components/resumption/src/last_state_impl.cc

```diff
@@ -42,34 +42,39 @@ LastStateImpl::LastStateImpl(const std::string& app_storage_folder,
                              const std::string& app_info_storage)
     : app_storage_folder_(app_storage_folder)
     , app_info_storage_(app_info_storage) {
-  LoadStateFromFileSystem();
+  LoadFromFileSystem();
   LOG4CXX_AUTO_TRACE(logger_);
 }
 
 LastStateImpl::~LastStateImpl() {
   LOG4CXX_AUTO_TRACE(logger_);
-  SaveStateToFileSystem();
+  SaveToFileSystem();
 }
 
-void LastStateImpl::SaveStateToFileSystem() {
+void LastStateImpl::SaveToFileSystem() {
   LOG4CXX_AUTO_TRACE(logger_);
-  const std::string& str = dictionary_.toStyledString();
-  const std::vector<uint8_t> char_vector_pdata(str.begin(), str.end());
 
+  std::string styled_string;
+  {
+    sync_primitives::AutoLock lock(dictionary_lock_);
+    styled_string = dictionary_.toStyledString();
+  }
+
+  const std::vector<uint8_t> char_vector_pdata(styled_string.begin(),
+                                               styled_string.end());
   DCHECK(file_system::CreateDirectoryRecursively(app_storage_folder_));
-  LOG4CXX_INFO(logger_,
-               "LastState::SaveStateToFileSystem " << app_info_storage_ << str);
+  LOG4CXX_INFO(
+      logger_,
+      "LastState::SaveToFileSystem " << app_info_storage_ << styled_string);
   DCHECK(file_system::Write(app_info_storage_, char_vector_pdata));
 }
 
-Json::Value& LastStateImpl::get_dictionary() {
```

Comment:
deprecated method without definition

Reply:
@jacobkeeler Done in 4fdae6f97e8445b4aebe5269c4163e028115f508

Reply:
This comment was not resolved

Reply:
@jacobkeeler The method is marked as `DEPRECATED` and the definition has been added.